modelforge/curation/qm9_curation.py

```diff
@@ -224,124 +232,143 @@ def _parse_xyzfile(self, file_name: str) -> dict:
         17  cal/mol/K   Heat capacity at 298.15K
 
         """
+        import qcelemental as qcel
+        from modelforge.utils.misc import str_to_float
 
         with open(file_name, "r") as file:
+            # read the first line that provides the number of atoms
             n_atoms = int(file.readline())
+
+            # the second line provides properties that we will parse into a dict
             properties_temp = file.readline()
             properties = self._parse_properties(properties_temp)
+
             elements = []
             atomic_numbers = []
             geometry = []
             charges = []
             hvf = []
+
+            geometry_temp = []
+            charges_temp = []
+
+            # loop over the atoms to get coordinates and charges
             for i in range(n_atoms):
                 line = file.readline()
                 element, x, y, z, q = line.split()
                 elements.append(element)
                 atomic_numbers.append(qcel.periodictable.to_atomic_number(element))
                 temp = [
-                    self._str_to_float(x),
-                    self._str_to_float(y),
-                    self._str_to_float(z),
+                    str_to_float(x),
+                    str_to_float(y),
+                    str_to_float(z),
                 ]
                 geometry.append(temp)
-                charges.append(self._str_to_float(q))
+                charges.append(str_to_float(q))
 
             hvf_temp = file.readline().split()
 
             smiles = file.readline().split()
             InChI = file.readline()
 
-            data = {}
-            data["name"] = file_name.split("/")[-1].split(".")[0]
-            data["smiles_gdb-17"] = smiles[0]
-            data["smiles_b3lyp"] = smiles[1]
-            data["inchi_Corina"] = InChI.split("\n")[0].split()[0].replace("InChI=", "")
-            data["inchi_B3LYP"] = InChI.split("\n")[0].split()[1].replace("InChI=", "")
-            data["geometry"] = np.array(geometry) * unit.angstrom
+            data_temp = {}
+            data_temp["name"] = file_name.split("/")[-1].split(".")[0]
+            data_temp["n_configs"] = 1
+            data_temp["smiles_gdb-17"] = smiles[0]
+            data_temp["smiles_b3lyp"] = smiles[1]
```

Comment:
is there a difference between the smiles strings?

Reply:
Yes, they can be; calculated using two different approaches I believe (which may have different assumptions that identify connectivity).